src/ipa_pc/mod.rs

```diff
@@ -736,6 +736,155 @@ impl<G: AffineCurve, D: Digest> PolynomialCommitment<G::ScalarField> for InnerPr
+        opening_challenges: &dyn Fn(u64) -> G::ScalarField,
+        rands: impl Clone + IntoIterator<Item = &'a Self::Randomness>,
+        rng: Option<&mut dyn RngCore>,
+    ) -> Result<Self::BatchProof, Self::Error>
+    where
+        Self::Randomness: 'a,
+        Self::Commitment: 'a,
+    {
+        let batch_time = start_timer!(|| "Multi poly multi point batching.");
+
+        let mut opening_challenge_counter = 0;
+        
+        // lambda
+        let mut cur_challenge = opening_challenges(opening_challenge_counter);
+        opening_challenge_counter += 1;
+
+        let mut query_to_labels_map = BTreeMap::new();
+
+        let poly_map: BTreeMap<_, _> = labeled_polynomials
+            .clone()
+            .into_iter()
+            .map(|poly| (poly.label(), poly))
+            .collect();
+
+        for (label, (point_label, point)) in query_set.iter() {
+            let labels = query_to_labels_map
+                .entry(point_label)
+                .or_insert((point, BTreeSet::new()));
+            labels.1.insert(label);
+        }
+
+        let mut points = vec![];
+
+        // h(X)
+        let mut batch_polynomial = Polynomial::zero();
+
+        for (_point_label, (point, labels)) in query_to_labels_map.into_iter() {
+
+            for label in labels {
+                let labeled_polynomial =
+                    poly_map.get(label).ok_or(Error::MissingPolynomial {
+                        label: label.to_string(),
+                    })?;
+
+                points.push(point);
+
+                // y_i
+                let evaluated_y = labeled_polynomial.polynomial().evaluate(*point); 
+
+                // (p_i(X) - y_i) / (X - x_i)
+                let polynomial = 
+                    &(labeled_polynomial.polynomial() - &Polynomial::from_coefficients_vec(vec![evaluated_y])) 
+                    / 
+                    &Polynomial::from_coefficients_vec(vec![
+                        (G::ScalarField::zero() - point), 
+                        G::ScalarField::one()
+                    ]);
+                
+                // h(X) = SUM( lambda^i * ((p_i(X) - y_i) / (X - x_i)) )
+                batch_polynomial += (cur_challenge, &polynomial);
+
+                // lambda^i
+                cur_challenge = opening_challenges(opening_challenge_counter);
+                opening_challenge_counter += 1;
+            }
+        }
+
+        // Commitment of the h(X) polynomial
+        let batch_commitment = Self::cm_commit(
+            ck.comm_key.as_slice(),
+            batch_polynomial.coeffs.as_slice(),
+            None,
+            None,
+        ).into_affine();
+
+        // TODO: When we will move to Sponge-based construction, we will absorb the batch_commitment only and then squeeze a new challenge
+        // Fresh random challenge x
+        let point= Self::compute_random_oracle_challenge(
+            &to_bytes![
+                batch_commitment,
+                points                
```

Comment:
Having the points (and other data) already absorbed by the random oracle ``opening_challenges()`` in the first step (as described in my comment above), we do not need to absorb them again. And we can use ``opening_challenges()`` instead of`` compute_random_oracle_challenge()``.